### PR TITLE
Remove bcl2fastq demultiplexing

### PR DESCRIPTION
**Pull request by diitaz93**
## Description
Remove the demultiplexing with bcl2fastq from cg.

### Changed

- Removed `bcl2fastq` sbatch command
- Made all demultiplexing jobs use BCLConvert in the DRAGEN
- Removed unused tests and fixtures

### How to prepare for test

- [x] Ssh to relevant server (depending on type of change)
- [x] Use stage: `us`
- [x] Paxa the environment: `paxa`
- [x] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b remove-bcl2fastq-03 -a
    ```

### How to test

- [x] `cg demultiplex all`, check that the flow cells are demultiplexed correctly
```shell
Slurm jobs are sent
demultiplexing jobs finished correctly on the DRAGEN
```

## Review

- [x] Tests executed by SD
- [x] "Merge and deploy" approved by CO
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MAJOR** - when you make incompatible API changes

## Implementation Plan

- [ ] Deployed to stage:
```shell

```
- [ ] Deployed to production:
```shell

```

### CODE REVIEW

File: cg/models/flow_cell/flow_cell.py [hunk 1]
```diff
@@ -249,7 +249,7 @@ def is_flow_cell_ready(self) -> bool:
             LOG.warning(f"Copy of sequence data is not ready for flow cell {self.id}")
             return False
         LOG.debug(f"All data has been transferred for flow cell {self.id}")
-        LOG.info(f"Flow cell {self.id} is ready for downstream processing")
+        LOG.info(f"Flow cell {self.id} is ready for demultiplexing or encrypting")
```
Comment (ChrOertlin) [inline cg/models/flow_cell/flow_cell.py:252]: I would keep downstream processing. Because what about backup, cleaning etc.?

**Review by ChrOertlin — APPROVED** [review]
Nice!

File: cg/apps/demultiplex/sbatch.py [hunk 2]
```diff
@@ -1,33 +1,6 @@
 # This one needs run_dir, out_dir, basemask and sample_sheet
 
-from cg.constants.demultiplexing import BclConverter
```
[the remark this answers is not in the document]
Comment (ChrOertlin) [reply cg/apps/demultiplex/sbatch.py:3]: _Major changes in software can include changes in the user interface, the introduction of new features, the removal of existing features_
from the web

File: cg/models/flow_cell/flow_cell.py (hunk 1 above)
Comment (diitaz93) [reply cg/models/flow_cell/flow_cell.py:252]: Ok I will change it 👍

**Pull request merged**